### PR TITLE
Add support for ESRT loading under Xen

### PR DESCRIPTION
**Pull request by DemiMarie**
This is needed for fwupd to work in Qubes OS.

### CODE REVIEW

**Comment by jevank** [comment]
Hi. I'm trying to test with[ Xen PR](https://github.com/QubesOS/qubes-vmm-xen/pull/135), but so far without success. I've enabled loglevel=7 but I can't even see the "esrt-init: loading" message

**Comment by jevank** [comment]
Okay, moving on, I'm on 5.4, and it needs to patch arch/x86/kernel/setup.c .

Now got
`esrt: Xen did not reserve ESRT, ignoring it`

**Comment by DemiMarie** [comment]
> Okay, moving on, I'm on 5.4, and it needs to patch arch/x86/kernel/setup.c .
> 
> Now I see `esrt: Xen did not reserve ESRT, ignoring it`

Yeah, same on my side.

File: 0001-Add-support-for-ESRT-loading-under-Xen.patch [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
++			"error %d\n", phys_addr, rc);
++		return false;
++	}
++
++	memset(&md, 0, sizeof(md));
++	md.attribute = info->mem.attr;
++	md.type = info->mem.type;
++	md.num_pages = info->mem.size >> XEN_PAGE_SIZE;
++	md.phys_addr = info->mem.addr;
++	return true;
```
Comment (jevank) [inline 0001-Add-support-for-ESRT-loading-under-Xen.patch:105]: memcpy(md_out, md, sizeof(md)); ?

Comment (DemiMarie) [reply 0001-Add-support-for-ESRT-loading-under-Xen.patch:104]: Oops… that would explain why it doesn’t work :rofl:

Now fixed.

**Comment by jevank** [comment]
Great, the next problem I see :)
`esrt: ESRT header doesn't fit on single memory map entry. (size: 16 max: 0)`

**Comment by DemiMarie** [comment]
> Great, the next problem I see :) `esrt: ESRT header doesn't fit on single memory map entry. (size: 16 max: 0)`

fixed

**Comment by jevank** [comment]
> fixed

I' still have the same message... :(

**Comment by DemiMarie** [comment]
> > fixed
> 
> I'm still have the same message... :(

Was misusing the Xen hypercall, whoops

**Comment by jevank** [comment]
Yeah, now it works! Thank you 🎉

**Comment by DemiMarie** [comment]
> Yeah, now it works! Thank you :tada:

You’re welcome!

**Comment by DemiMarie** [comment]
Closing as I will use a different patch.

**Pull request closed without merging**